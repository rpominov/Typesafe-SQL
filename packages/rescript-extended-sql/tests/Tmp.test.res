open Jest

test("Simplest query", () => {
  expect(Parser.parse("SELECT 1"))->toMatchSnapshot
})
