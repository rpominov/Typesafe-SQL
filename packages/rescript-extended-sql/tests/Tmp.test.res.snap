// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block comments 1`] = `
Object {
  "TAG": 0,
  "_0": Object {
    "ast": Array [
      Object {
        "TAG": 3,
        "_0": " test 
 abc",
      },
      Object {
        "TAG": 0,
        "_0": "
SELECT ",
      },
      Object {
        "TAG": 3,
        "_0": " test2 /* test3 */ ",
      },
      Object {
        "TAG": 0,
        "_0": " 1",
      },
      Object {
        "TAG": 3,
        "_0": "",
      },
    ],
    "attributes": Object {
      "name": undefined,
    },
    "rawText": "/* test 
 abc*/
SELECT /* test2 /* test3 */ */ 1/**/",
  },
}
`;

exports[`Block comments error 1`] = `
Object {
  "TAG": 1,
  "_0": Object {
    "message": "Was expecting a block comment close sequence */, but reached the end of the string",
    "pos": 15,
  },
}
`;

exports[`Block comments error nested 1`] = `
Object {
  "TAG": 1,
  "_0": Object {
    "message": "Was expecting a block comment close sequence */, but reached the end of the string",
    "pos": 25,
  },
}
`;

exports[`Inline comments 1`] = `
Object {
  "TAG": 0,
  "_0": Object {
    "ast": Array [
      Object {
        "TAG": 2,
        "_0": " test",
      },
      Object {
        "TAG": 0,
        "_0": "SELECT ",
      },
      Object {
        "TAG": 2,
        "_0": " test2",
      },
      Object {
        "TAG": 2,
        "_0": "",
      },
      Object {
        "TAG": 0,
        "_0": " 1",
      },
      Object {
        "TAG": 2,
        "_0": " test3",
      },
    ],
    "attributes": Object {
      "name": undefined,
    },
    "rawText": "-- test
SELECT -- test2
--
 1-- test3",
  },
}
`;

exports[`Name attribute (2nd ignored) 1`] = `
Object {
  "TAG": 0,
  "_0": Object {
    "ast": Array [
      Object {
        "TAG": 3,
        "_0": "@name: test
@name: test1",
      },
      Object {
        "TAG": 0,
        "_0": "SELECT 1",
      },
    ],
    "attributes": Object {
      "name": "test",
    },
    "rawText": "/*@name: test
@name: test1*/SELECT 1",
  },
}
`;

exports[`Name attribute (empty) 1`] = `
Object {
  "TAG": 1,
  "_0": Object {
    "message": "Invalid @name attribute: ",
    "pos": -1,
  },
}
`;

exports[`Name attribute (invalid) 1`] = `
Object {
  "TAG": 0,
  "_0": Object {
    "ast": Array [
      Object {
        "TAG": 3,
        "_0": "@name: %abc",
      },
      Object {
        "TAG": 0,
        "_0": "SELECT 1",
      },
    ],
    "attributes": Object {
      "name": "%abc",
    },
    "rawText": "/*@name: %abc*/SELECT 1",
  },
}
`;

exports[`Name attribute 1`] = `
Object {
  "TAG": 0,
  "_0": Object {
    "ast": Array [
      Object {
        "TAG": 3,
        "_0": "@name: test",
      },
      Object {
        "TAG": 0,
        "_0": "SELECT 1",
      },
    ],
    "attributes": Object {
      "name": "test",
    },
    "rawText": "/*@name: test*/SELECT 1",
  },
}
`;

exports[`Simplest query 1`] = `
Object {
  "TAG": 0,
  "_0": Object {
    "ast": Array [
      Object {
        "TAG": 0,
        "_0": "SELECT 1",
      },
    ],
    "attributes": Object {
      "name": undefined,
    },
    "rawText": "SELECT 1",
  },
}
`;
